salt/modules/cmdmod.py

```diff
@@ -245,6 +245,10 @@ def _run(cmd,
     '''
     Do the DRY thing and only call subprocess.Popen() once
     '''
+    # Remove whitespace from strings
+    if isinstance(cmd, six.string_types):
+        cmd = cmd.strip()
+
```

Comment:
I'll double-check on the shell injection testing, if it's true then we can leave it in and then document it with a comment so that someone else doesn't see it and get the same idea I did and try to remove it.